src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1449,7 +1449,7 @@ private void readGroupExpression(Macro.Parameter parameter, List<Expression.EExp
             int startIndex = expressions.size();
             expressions.add(Expression.Placeholder.INSTANCE);
             boolean isSingleton = true;
-            while (nextGroupedValue() != Event.NEEDS_INSTRUCTION) {
+            while (nextGroupedValue() != Event.NEEDS_INSTRUCTION || isMacroInvocation()) {
```

Comment:
NEEDS_INSTRUCTION is used either when the end of the group has been reached, or when a macro invocation header has been encountered. When it's the latter, we need to read the invocation as an expression to continue progressing through the group.